### DISCUSSION TITLE
i think colors have too many "bits" (24)

### DISCUSSION THREAD

**Discussion by atesin** [nesbox/TIC-80#2195] [Ideas]
hi.... i see no need for colors to have so much depth in a retro console...  i remember i was happy with atari (16 colors palette from 256 different colors) or msdos vga games (320x200 with 256 simultaneous colors)

these bytes could better distributed: the current palette with 16 colors of 3 bytes is 48 bytes total (https://github.com/nesbox/TIC-80/wiki/RAM#palette)

... wouldn't be better with same 48 bytes, a palette with 48 colors of 1 byte?? =D  ... 8bit 332 rgb format could be used for this, is easy and looks good

https://en.wikipedia.org/wiki/List_of_software_palettes#8-8-4_levels_RGB

**Comment by Anrock** [comment]
There is a Bits Per Pixel switch in sprite editor after you enable advanced mode via toggle in top left corner

**Reply by atesin** [reply]
yes i know, but still not the same.... 

with these switches (visible with advanced mode enabled) you can only select the size of palette (16, 4 or 2 simultaneous colors), but you can still select those 16, 4 or 2 colors from a set of 24 millions (3 bytes) possible colors through the three RGB component sliders.....  with 1 byte we can get 256 possible colors but more simultaneous ones (being 48, 12 or 6 changing bpp, what you are saying)

is not hard to understand... would you like a 3x bigger palette??

**Comment by joshgoebel** [comment]
I don't really see this request having any traction, this is foundational part of TIC-80.... 16 colors, from a 24-bit palette.  If you don't want to use that full colorspace in your own game - then don't - that's up to you.

**Comment by atesin** [comment]
hi @joshgoebel  ... i was thinking a bit and i was wrong, lowering the color bits wont bring us a larger palette, that has to do with sprite and framebuffer vram ...  

i think sprites and framebuffer have 2 pixels per byte, and each nibble (half byte) points to some of the 16 palette color, it matches perfectly...  256 simultaneous colors (1 pixel per byte) could be done but that would mean doubling framebuffer size (and/or halfing sprite space....  maybe an interesting option for future releases, to add simultaneous 8bpp support at cost of half vram :) )

**Reply by joshgoebel** [reply]
I don't see much likelihood of that happening either.  16 colors was/is an intentional "hardware" design restriction.

**Reply by atesin** [reply]
i see

i am thinking... i am sure you always receive comments suggesting (or demanding) expand hardware capabilities of the computer, like "why dont you increase screen resolution?.. why dont you add more sound channels?"...  etc

i realize that is a retro computer, a computer that is constrained on resources, because were most possible at that times...  that is how the computers were before, and programmers had to use their creativity to make things work, and still they were able to create masterpieces that we even remember and appreciate until today, because have a a different feeling

many modern AAA games are just stunning graphics, but have no soul.... you note many of them are still bored, or addictive, or greedy
